tools/config.schema.yaml

```diff
@@ -0,0 +1,855 @@
+                  MESSAGES_TYPES:
+                    type: array
+                    uniqueItems: True
+                    items:
+                      type: string
+                      enum: ["received", "sent"]
+                  FEATURES:
+                    type: object
+                    required: [received, sent]
+                    properties:
+                      received:
+                        type: array
+                        uniqueItems: True
+                        items:
+                          type: string
+                          enum: ["count", "distinctcontacts", "timefirstmessage", "timelastmessage", "countmostfrequentcontact"]
+                      
+                      sent:
+                        type: array
+                        uniqueItems: True
+                        items:
+                          type: string 
+                          enum: ["count", "distinctcontacts", "timefirstmessage", "timelastmessage", "countmostfrequentcontact"]
+    
+  PHONE_SCREEN:
+    type: object
+    required: [TABLE, PROVIDERS]
+    properties:
+      TABLE: 
+        type: string
+      PROVIDERS:
+        type: ["null", object]
+        properties:
+          RAPIDS:
+            allOf:
+              - $ref: "#/definitions/PROVIDER"
+              - properties:
+                  REFERENCE_HOUR_FIRST_USE: 
+                    type: integer
+                  IGNORE_EPISODES_SHORTER_THAN:
+                    type: integer
+                    minimum: 0
+                  IGNORE_EPISODES_LONGER_THAN:
+                    type: integer
+                    minimum: 0
+                  FEATURES:
+                    uniqueItems: True
+                    items:
+                      type: string
+                      enum: ["countepisode", "sumduration", "maxduration", "minduration", "avgduration", "stdduration", "firstuseafter"] 
+                  EPISODE_TYPES:
+                    type: array
+                    uniqueItems: True
+                    items:
+                      type: string
+                      enum: ["unlock"]
+
+  PHONE_WIFI_CONNECTED:
+    type: object
+    required: [TABLE, PROVIDERS]
+    properties:
+      TABLE: 
+        type: string
+      PROVIDERS:
+        type: ["null", object]
+        properties:
+          RAPIDS:
+            allOf:
+              - $ref: "#/definitions/PROVIDER"
+              - properties:
+                  FEATURES:
+                    items:
+                      type: string
+                      enum: ["countscans", "uniquedevices", "countscansmostuniquedevice"]
+        additionalProperties: 
+          $ref: "#/definitions/PROVIDER"
+  
+  PHONE_WIFI_VISIBLE:
+    type: object
+    required: [TABLE, PROVIDERS]
+    properties:
+      TABLE:
+        type: string
+      PROVIDERS:
+        type: ["null", object]
+        properties:
+          RAPIDS:
+            allOf:
+              - $ref: "#/definitions/PROVIDER"
```

Comment:
Missing the same feature validations as PHONE_WIFI_CONNECTED

Reply:
fixed

Reply:
still missing the feature validations